fastlane/lib/fastlane/actions/register_devices.rb

```diff
@@ -54,6 +48,13 @@ def self.run(params)
         return device_objs
       end
 
+      def try_create_device(name, udid, mac)
+        Spaceship::Device.create!(name: name, udid: udid, mac: mac)
+      rescue => ex
+        UI.error("Failed to register new device (name #{name}, UDID: #{udid})")
+        UI.error(ex.to_s)
```

Comment:
I think we might want to make this...
```rb
UI.error(ex.to_s)
UI.crash!("Failed to register new device (name #{name}, UDID: #{udid})")
```

We want to make sure that we end execution and that an error return code gets bubbled to the top of this execution